envoy/server/filter_config.h

```diff
@@ -262,8 +262,8 @@ class NamedHttpFilterConfigFactory : public virtual HttpFilterConfigFactoryBase
    * @return Http::FilterFactoryCb the factory creation function.
    */
   virtual Http::FilterFactoryCb
-  createFilterServerFactoryFromProto(const Protobuf::Message&, const std::string&,
-                                     Server::Configuration::ServerFactoryContext&) {
+  createFilterFactoryFromProto(const Protobuf::Message&, const std::string&,
```

Comment:
Yea, because of build pain like gcc failure. I chosen just give it a new name instead of function overloading before

So how about just rename it from `createFilterServerFactoryFromProto` to `createFilterFactoryFromProtoWithServerContext` like you have. 
By doing this, we can 1) solve your gcc build issue 2) avoid redundant override in `DualFactoryBase` because it doesn't support at this moment anyways and 3)also achieve the purpose of clarifying the function name. 4)It also solve your http/common coverage issue. These benefits seem to outweigh the unified name.

Reply:
sgtm